In [1]:
import numpy as np
import pandas as pd 
import requests 
from bs4 import BeautifulSoup

In [2]:
requests.get('https://www.ambitionbox.com/list-of-companies?page=1')

<Response [403]>

'''
If it shows Responses 403 the use this :
headers={'User-Agent':'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.36'} 
requests.get('url',headers=headers).text
'''

In [3]:
headers={'User-Agent':'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.36'} 
webpage = requests.get('https://www.ambitionbox.com/list-of-companies?page=1',headers=headers).text

In [4]:
# making the Soup object 
soup = BeautifulSoup(webpage ,'lxml')

In [5]:
# find the 1 header and extracting text
soup.find_all('h1')[0].text.strip()

'Top Companies in\n\t\t\t\t \n\t\t\t\t\tINDIA'

In [6]:
for i in soup.find_all('h2'):
    print(i.text.strip())

Companies in India
TCS
Accenture
Wipro
Cognizant
Capgemini
HDFC Bank
Infosys
ICICI Bank
HCLTech
Tech Mahindra
Genpact
TP
Jio
Axis Bank
Concentrix Corporation
Amazon
Reliance Retail
iEnergizer
LTIMindtree
IBM
Popular Collections by Industries
Popular Collections by Cities
Popular Collections by Roles


# To consider the whole container

In [7]:
company = soup.find_all('div' ,class_='companyCardWrapper')

In [8]:
len(company)

20

In [22]:
names= []
ratings = []
reviews = []
# ctype=[]
# hq=[]
# how_old=[]
# no_of_employee=[]

for i in company:
    names.append(i.find('h2').text.strip())
    ratings.append(i.find('div',class_='rating_text').text.strip())
    reviews.append(i.find('a',class_='companyCardWrapper__ActionWrapper').text.strip())
    
d = {'name': names,'ratings': ratings,'reviews':reviews}

df = pd.DataFrame(d)

In [23]:
df.head()

,name,ratings,reviews
0,TCS,3.3,1.1L Reviews
1,Accenture,3.7,73k Reviews
2,Wipro,3.6,64.7k Reviews
3,Cognizant,3.7,61k Reviews
4,Capgemini,3.7,52.9k Reviews


In [26]:
# Its of only 1 pages = 20 compaines
df.shape

(20, 3)

Final database (getting all pages scrap)

In [34]:
final = pd.DataFrame()

for j in range(1, 11):
    headers={'User-Agent':'Mozilla/5.0 (Windows NT 6.3; Win 64 ; x64) Apple WeKit /537.36(KHTML , like Gecko) Chrome/80.0.3987.162 Safari/537.36'} 
    webpage = requests.get(f'https://www.ambitionbox.com/list-of-companies?page={j}',headers=headers).text
    soup = BeautifulSoup(webpage, 'lxml')
    company = soup.find_all('div', class_='companyCardWrapper')

    names = []
    ratings = []
    reviews = []

    for i in company:
        names.append(i.find('h2').text.strip())

        rating_tag = i.find('div', class_='rating_text')
        ratings.append(rating_tag.text.strip() if rating_tag else None)

        review_tag = i.find('a', class_='companyCardWrapper__ActionWrapper')
        reviews.append(review_tag.text.strip() if review_tag else None)

    d = {'name': names, 'ratings': ratings, 'reviews': reviews}
    df = pd.DataFrame(d)

    final = pd.concat([final, df], ignore_index=True)

In [36]:
final.head()

,name,ratings,reviews
0,TCS,3.3,1.1L Reviews
1,Accenture,3.7,73k Reviews
2,Wipro,3.6,64.7k Reviews
3,Cognizant,3.7,61k Reviews
4,Capgemini,3.7,52.9k Reviews


In [38]:
print(f"1 page =",df.shape)
print(f"10 page = ",final.shape)

1 page = (20, 3)
10 page =  (200, 3)
